src/azure-cli/azure/cli/command_modules/acr/_params.py

```diff
@@ -316,6 +316,18 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('password1', options_list=['--password1'], help='Flag indicating if first password should be deleted', action='store_true', required=False)
         c.argument('password2', options_list=['--password2'], help='Flag indicating if second password should be deleted.', action='store_true', required=False)
 
+    with self.argument_context('acr agentpool') as c:
+        c.argument('registry_name', options_list=['--registry', '-r'])
+        c.argument('agentpool_name', options_list=['--name', '-n'], help='The name of the agentpool.')
+        c.argument('tag', help='The tag of the agentpool.')
+        c.argument('count', type=int, help='The count of the agentpool.')
+        c.argument('tier', help='The tier of the agentpool.')
```

Comment:
shouldn't we make a data model for the agent pool tiers just like ACR has for the SKU?

Reply:
My assumption is that SKU is an Enum, and ACR does not model it as string. Downside of Enums are if they want to add any new SKU's they will have to go thru new API version. @northtyphoon can correct me if I am wrong

Reply:
I know that is the downside, but if they already have the SKU like this which is like our tier I think is better to be consistent. I doubt we will be changing the tiers very often anyway
it is an enum
`        c.argument('sku', help='The SKU of the container registry', arg_type=get_enum_type(SkuName))`

and a data model 
`SkuName, PasswordName, DefaultAction, PolicyStatus, WebhookAction, WebhookStatus, TaskStatus, \
        BaseImageTriggerType, RunStatus, SourceRegistryLoginMode, UpdateTriggerPayloadType, TokenStatus = self.get_models(
            'SkuName', 'PasswordName', 'DefaultAction', 'PolicyStatus', 'WebhookAction', 'WebhookStatus',
            'TaskStatus', 'BaseImageTriggerType', 'RunStatus', 'SourceRegistryLoginMode', 'UpdateTriggerPayloadType',
            'TokenStatus')
`